### PR TITLE
Show all options when clicking arrow in autocomplete form - Mct4578

### PR DESCRIPTION
**Pull request by scottbell**
resolves: #4578

### All Submissions:

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [X] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [X] Changes address original issue?
* [ ] Unit tests included and/or updated with changes?
* [X] Command line build passes?
* [X] Has this been smoke tested?
* [X] Testing instructions included in associated issue?

### CODE REVIEW

**Comment by scottbell** [comment]
This addresses:
https://github.com/nasa/openmct/issues/4578

**Comment by unlikelyzero** [comment]
@scottbell it looks like we have pretty terrible coverage for this component. Do you have some thoughts on how much effort it would take to add a single test to cover this component.

**Comment by scottbell** [comment]
> @scottbell it looks like we have pretty terrible coverage for this component. Do you have some thoughts on how much effort it would take to add a single test to cover this component.

I think @nikhilmandlik has a ticket to add tests to cover all the form API components.

File: src/api/forms/components/controls/AutoCompleteField.vue [hunk 1]
```diff
@@ -155,11 +165,14 @@ export default {
         },
         inputClicked() {
             this.autocompleteInputElement.select();
-            this.showOptions(this.autocompleteInputElement.value);
+            this.showOptions();
         },
         arrowClicked() {
+            // if the user clicked the arrow, we want
+            // to show them all the options
+            this.showFilteredOptions = false;
```
Comment (nikhilmandlik) [inline src/api/forms/components/controls/AutoCompleteField.vue:173]: 👍

File: src/api/forms/components/controls/AutoCompleteField.vue [hunk 2]
```diff
@@ -155,11 +165,14 @@ export default {
         },
         inputClicked() {
             this.autocompleteInputElement.select();
-            this.showOptions(this.autocompleteInputElement.value);
+            this.showOptions();
```
Comment (nikhilmandlik) [inline src/api/forms/components/controls/AutoCompleteField.vue:168]: may be set `this.showFilteredOptions = true;`?

File: src/api/forms/components/controls/AutoCompleteField.vue [hunk 3]
```diff
@@ -155,11 +165,14 @@ export default {
...
         },
         arrowClicked() {
+            // if the user clicked the arrow, we want
+            // to show them all the options
+            this.showFilteredOptions = false;
             this.autocompleteInputElement.select();
-            this.showOptions('');
+            this.showOptions();
```
Comment (nikhilmandlik) [inline src/api/forms/components/controls/AutoCompleteField.vue:175]: wondering if we really need to call `this.showOptions();` (same on inputClicked)  I think reactive `showFilteredOptions` change takes care of `filteredOptions`? Can you confirm?

**Comment by nikhilmandlik** [comment]
Hey @scottbell little curious about changes in reactive data `showFilteredOptions` triggering filterOption, if that doesn't work changes looks good 👍

**Comment by scottbell** [comment]
To test:

- Create a Clock Object, observe the timezone dropdown
- Ensure you can click the arrow in the timezone options and display all timezones
- Type some text and ensure it filters the timezones appropriately

**Comment by akhenry** [comment]
@nikhilmandlik @scottbell  This good to go?

**Comment by scottbell** [comment]
> @nikhilmandlik @scottbell This good to go?

From my perspective, I think so

**Pull request merged**